exe/slow_query_exporter

```diff
@@ -1,78 +1,122 @@
+    # iteration. We may get _less_ data than this, but we'll never get _more_ data than this. The data read may contain
+    # any number of complete lines (anywhere from 0 to N).
+    CHUNK_SIZE = 8192
 
-$gelf_host = "localhost"
-$gelf_port = 12201
-$delay = 0.1
-$verbose = false
-$foreground = false
+    def initialize
+      log4r_config = File.join(Gem::Specification.find_by_name("slow_query_exporter").gem_dir, "config", "log4r.yaml")
+      Log4r::YamlConfigurator["IDENT"] = File.basename($0)
+      Log4r::YamlConfigurator.load_yaml_file(log4r_config)
+      self.logger = Log4r::Logger["sqe"]
+      self.logger.level = Log4r::INFO
+      super
 
-opts = OptionParser.new do |opts|
-  opts.banner = "Usage: slow_query_exporter [-fv] [-d delay] [-h host] [-p port] /path/to/slow_query.log"
-  opts.version = SlowQueryExporter::VERSION
-  opts.summary_width = 22
-  opts.on("-d SECONDS", "--delay SECONDS", Float, "An interval to pause after each GELF message, in (possibly fractional) seconds (default: #{$delay}s)") { |delay| $delay = delay.to_f }
-  opts.on("-f", "--foreground", "Don't daemonize on startup") { $foreground = true }
-  opts.on("-?", "--help", "Display this help text") do
-    puts(opts)
-    exit
-  end
-  opts.on("-h HOST", "--host HOST", "The GELF host (default: #{$gelf_host})") { |host| $gelf_host = host }
-  opts.on("-p PORT", "--port PORT", Integer, "The GELF UDP port (default: #{$gelf_port})") { |port| $gelf_port = port }
-  opts.on("-v", "--[no-]verbose", "Print entries to stdout as they're parsed") { $verbose = true }
-  opts.on("--version", "Display application version information") do
-    puts(opts.version)
-    exit
-  end
-end
-begin
-  opts.parse!
-  raise OptionParser::MissingArgument.new("/path/to/slow_query.log is required") if ARGV.empty?
-  raise OptionParser::InvalidArgument.new("#{ARGV.slice(1, ARGV.size).join(' ')}") if ARGV.size > 1
-rescue OptionParser::InvalidOption, OptionParser::InvalidArgument, OptionParser::MissingArgument => e
-  $stderr.puts(e)
-  $stderr.puts
-  $stderr.puts(opts)
-  exit 2
-end
+      option_parser.banner = "#{File.basename($0)} [-dv] [--sleep delay] [-h host] [-p port] [options] /path/to/slow_query.log"
+      option_parser.version = SlowQueryExporter::VERSION
+      option_parser.summary_width = 22
 
-Daemons.daemonize(ontop: $foreground, app_name: "slow_query_exporter", dir_mode: :normal, dir: RUN_DIRECTORY)
-Syslog.open("slow_query_exporter", Syslog::LOG_PID | Syslog::LOG_PERROR, Syslog::LOG_DAEMON)
-last_timestamp = begin
-                   IO.read(PROGRESS_FILE).to_i
-                 rescue Errno::ENOENT
-                   0
-                 end
+      options[:gelf_host] ||= "127.0.0.1"
+      option_parser.on(
+        "-h HOST", "--host HOST",
+        "target GELF host (default: #{options[:gelf_host]})"
+      ) do |host|
+        begin
+          Resolv::DNS.new.getaddress(host)
```

Comment:
@fimmtiu Oh, yeah. Socket.gethostbyname would work as well. I was just thinking "DNS resolution -> must use Resolv". Could change to `Socket` and not use `Resolv`.

As you say, `Resolv` is part of the stdlib though, so not a huge difference I think.